### PR TITLE
Update Network ProviderNamespace to new RBAC structure, input params & child-resource ref

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Update Network ProviderNamespace to new RBAC structure, input params & child-resource ref

Parameter reference
[Network: Applicationgateways](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.applicationgateways.yml)
[Network: Applicationsecuritygroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.applicationsecuritygroups.yml)
[Network: Azurefirewalls](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.azurefirewalls.yml)
[Network: Bastionhosts](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.bastionhosts.yml)
[Network: Ddosprotectionplans](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.ddosprotectionplans.yml)
[Network: Expressroutecircuits](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.expressroutecircuits.yml)
[Network: Ipgroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.ipgroups.yml)
[Network: Loadbalancers](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.loadbalancers.yml)
[Network: Localnetworkgateways](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.localnetworkgateways.yml)
[Network: Natgateways](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.natgateways.yml)
[Network: Networksecuritygroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.networksecuritygroups.yml)
[Network: Networkwatchers](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.networkwatchers.yml)
[Network: Privateendpoints](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.privateendpoints.yml)
[Network: Publicipaddresses](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.publicipaddresses.yml)
[Network: Publicipprefixes](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.publicipprefixes.yml)
[Network: Routetables](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.routetables.yml)
[Network: Trafficmanagerprofiles](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.trafficmanagerprofiles.yml)
[Network: Virtualnetworkgateways](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.virtualnetworkgateways.yml)
[Network: Virtualnetworks](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.virtualnetworks.yml)
[Network: Virtualwans](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.virtualwans.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: arm/Microsoft.Network/networkWatchers/connectionMonitors/readme.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ This template deploys Connection Monitors.
 | `networkWatcherName` | string | `[format('NetworkWatcher_{0}', resourceGroup().location)]` |  | Optional. Name of the network watcher resource. Must be in the resource group where the Flow log will be created and same region as the NSG |
 | `tags` | object | `{object}` |  | Optional. Tags of the resource. |
 | `testConfigurations` | array | `[]` |  | Optional. List of connection monitor test configurations. |
-| `testGroups` | array | `[]` |  | Optional. List of connection monitor test groups. |
+| `testGroups` | array | `[]` |  | Optional.	List of connection monitor test groups. |
```
Comment (rahalan) [inline arm/Microsoft.Network/networkWatchers/connectionMonitors/readme.md:22]: too many spaces

File: arm/Microsoft.Network/virtualNetworks/readme.md [hunk 2]
```diff
@@ -154,8 +154,8 @@ The network security group and route table resources must reside in the same res
 
 | Output Name | Type | Description |
 | :-- | :-- | :-- |
-| `subnetIds` | array | The resourceIds of the deployed subnets |
-| `subnetNames` | array | The names of the deployed subnets |
+| `subnetIds` | array |  |
```
Comment (rahalan) [inline arm/Microsoft.Network/virtualNetworks/readme.md:157]: description is missing

File: arm/Microsoft.Network/networkWatchers/connectionMonitors/readme.md (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.Network/networkWatchers/connectionMonitors/readme.md]: done

File: arm/Microsoft.Network/virtualNetworks/readme.md (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.Network/virtualNetworks/readme.md]: done

**Review by rahalan — APPROVED** [review]

**Pull request merged**